src/repository.ts

```diff
@@ -1,78 +1,82 @@
 import { FeatureConfig, Segment } from './openapi';
-import { KeyValueStore, Query } from './types';
+import { AsyncKeyValueStore, KeyValueStore, Query } from './types';
 
 export interface Repository extends Query {
   // put values
-  setFlag(identifier: string, fc: FeatureConfig): void;
-  setSegment(identifier: string, segment: Segment): void;
+  setFlag(identifier: string, fc: FeatureConfig): Promise<void>;
+  setSegment(identifier: string, segment: Segment): Promise<void>;
 
   // remove values
-  deleteFlag(identifier: string): void;
-  deleteSegment(identifier: string): void;
+  deleteFlag(identifier: string): Promise<void>;
+  deleteSegment(identifier: string): Promise<void>;
 }
 
 export class StorageRepository implements Repository {
   private cache: KeyValueStore;
-  private store: KeyValueStore;
+  private store: AsyncKeyValueStore;
 
-  constructor(cache: KeyValueStore, store?: KeyValueStore) {
+  constructor(cache: KeyValueStore, store?: AsyncKeyValueStore) {
     if (!cache) {
-      throw new Error('Cache is required argument and connot be undefined');
+      throw new Error('Cache is required argument and cannot be undefined');
     }
     this.cache = cache;
     this.store = store;
   }
 
-  setFlag(identifier: string, fc: FeatureConfig): void {
+  async setFlag(identifier: string, fc: FeatureConfig): Promise<void> {
     const flagKey = this.formatFlagKey(identifier);
-    if (this.isFlagOutdated(flagKey, fc)) {
+    if (await this.isFlagOutdated(flagKey, fc)) {
       return;
     }
     if (this.store) {
-      this.store.set(flagKey, fc);
+      await this.store.set(flagKey, fc);
       this.cache.del(flagKey);
     } else {
       this.cache.set(flagKey, fc);
     }
+    return;
```

Comment:
Since you're not returning early or returning anything, you don't need trailing `return`'s like this.

Reply:
Done